### DISCUSSION TITLE
LXC consoles function keys

### DISCUSSION THREAD

**Discussion by nagyrobi** [tteck/Proxmox#973] [Ideas]
_TLDR_
@tteck Thank you for your wonderful scripts these are awesome and increase my productivity _a lot_. I've migrated everything from ESXi to Proxmox and taking that decision was empowered by the existence of these scripts, which made me realize how better and simpler life is by using them. I realized I don't need a lot of VMs, since with LXCs I gain resources and performance and robustness.

I run into [this](https://forum.proxmox.com/threads/function-keys-with-lxc-and-xterm-js.61747/) with every LCX container installer script. The behavior is not consistent between Proxmox's own shell and LXC container's own shell, function keys (F1-F12) don't work.

Use case is for example using `mc` in the console which relies on many function keys for basic operations.

The workaround suggested in the forum is to add `export TERM='xterm-256color'` to `.bashrc`. Perhaps it would be a good idea to add it by default to every container created with your scripts.

**Comment by tteck** [comment]
Interesting, `mc` isn't installed by default. Maybe I should include Midnight Commander with `export TERM='xterm-256color'` work around?
Meanwhile, you can run `echo "export TERM='xterm-256color'" >>/root/.bashrc` in each LXC console

**Reply by nagyrobi** [reply]
`mc` isn't installed by default but this is not specific to `mc`. There can by any console application requiring function keys usage (various text editors, serial terminal emulators etc) so I'd rather universally apply it by default.

> Meanwhile, you can run `echo "export TERM=xterm-256color" >>/root/.bashrc` in each LXC console

Yeah, done that, just a suggestion for the future...

**Reply by tteck** [reply]
Done
